app/models/katello/sync_plan.rb

```diff
@@ -36,6 +37,38 @@ def product_enabled
       end
     end
 
+    def save_with_logic!
+      associate_recurring_logic
+      self.save!
+      start_recurring_logic
+    end
+
+    def update_attributes_with_logics!(params)
+      self.update_attributes!(params)
+      if rec_logic_changed?
+        old_rec_logic = self.foreman_tasks_recurring_logic
+        associate_recurring_logic
+        self.save!
+        old_rec_logic.cancel
+        start_recurring_logic
+      end
+      toggle_enabled if enabled_toggle?
+    end
+
+    def associate_recurring_logic
+      self.foreman_tasks_recurring_logic = add_recurring_logic_to_sync_plan(self.sync_date, self.interval)
+    end
+
+    def toggle_enabled
+      self.foreman_tasks_recurring_logic.enabled = self.enabled
+    end
+
+    def start_recurring_logic
```

Comment:
@jlsherrill : There was a silent error related to permissions here..Does this look Ok?

Reply:
The User.as anonymous admin do part..